src/core/include/units/math.h

```diff
@@ -187,6 +187,21 @@ template<Unit To, typename D, typename U, typename Rep>
   }
 }
 
+/**
+ * @brief Overload of @c ::units::floor<Unit>() using the unit type of To
+ *
+ * @tparam q Quantity being the base of the operation
+ * @return Quantity The rounded quantity with unit type of quantity To
+ */
+template<Quantity To, typename D, typename U, typename Rep>
+[[nodiscard]] constexpr quantity<D, typename To::unit, Rep> floor(const quantity<D, U, Rep>& q) noexcept
```

Comment:
We should ensure that `To` has the same dimension and representation as `quantity<D, U, Rep>`.

Reply:
Isn't that subsumed by the _simple-requirement_ `::units::floor<typename To::unit>(q);`?

Reply:
@mpusz

Reply:
I don't think so. Assume a quantity with integer representation and you specify the target type with floating point. I guess there it makes sense to stop as we cannot really know which representation the user would expect.

Reply:
In that case, the requirement `std::same_as<typename To::dimension, D>` is subsumed by `::units::floor<typename To::unit>(q);`, right?

Reply:
For the dimension I was not sure. Here I just followed the suggestion of @mpusz. Maybe he can clarify.